### PR TITLE
Added CSV export.

### PR DESCRIPTION
**Pull request by marc**

### CODE REVIEW

File: spec/controllers/observation_exports_controller_spec.rb [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+RSpec.describe ObservationExportsController, type: :controller do
+  describe 'GET #index with a valid user' do
+    login_user
+    it 'returns http success and csv data' do
+      user = User.new(
+        email: 'user@localhost',
+        password: 'password'
+      )
+      os = ObservationSession.create( user: user, id: SecureRandom.uuid )
```
Comment (rposborne) [inline spec/controllers/observation_exports_controller_spec.rb]: Will the db set a UUID if we don't pass one from ruby?

File: app/models/observation.rb [hunk 2]
```diff
@@ -1,3 +1,15 @@
 class Observation < ApplicationRecord
   belongs_to :observation_session
+
+  comma do
+    id "id"
+    observation_session_id 'observation_session_id'
+    observation_session :user_id => 'user_id'
+    observation_session 'user_email' do |os| os.user&.email end
```
Comment (rposborne) [inline app/models/observation.rb]: Nit, lets do curly {} blocks over do an end for a single line code block.

**Review by rposborne — CHANGES_REQUESTED** [review]

File: app/controllers/observation_exports_controller.rb [hunk 3]
```diff
@@ -0,0 +1,17 @@
+class ObservationExportsController < ApplicationController
+  before_action :authenticate!
```
Comment (rposborne) [inline app/controllers/observation_exports_controller.rb]: I bet devise has a way to do this for us.

File: app/controllers/observation_exports_controller.rb [hunk 4]
```diff
@@ -0,0 +1,10 @@
+class ObservationExportsController < ApplicationController
+  before_action :authenticate_user!
```
Comment (kwokster10) [inline app/controllers/observation_exports_controller.rb]: Can we move this to the `ApplicationController`?

**Comment by kwokster10** [comment]
👍

**Pull request merged**